jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererCodeCoverageTracker.kt

```diff
@@ -60,13 +59,10 @@ abstract class CodeWhispererCodeCoverageTracker(
     }
 
     fun documentChanged(event: DocumentEvent) {
-        // When open a file for the first time, IDE will also emit DocumentEvent for loading with `isWholeTextReplaced = true`
-        // Added this condition to filter out those events
+        // Added this condition to filter out IDE reloading files
         if (event.isWholeTextReplaced) {
             LOG.debug { "event with isWholeTextReplaced flag: $event" }
-            (event as? DocumentEventImpl)?.let {
-                if (it.initialStartOffset == 0 && it.initialOldLength == event.document.textLength) return
-            }
+            if (event.oldTimeStamp == 0L) return
```

Comment:
Is it guaranteed that, for all the reloading documentChanged events, oldTimeStamp is 0, and for all the normal user paste cases, this value is non-zero?

Reply:
>> for all the normal user paste cases 
yes, they will have timestamp different from 0.

>> for all reloading documentChanged events
yes, from what I've observed
- case1: IDE reloading users' change on the document -> timestamp will be 0
- case2: some times IDE will reload the welcome.md (i can't consistent trigger this event), normally it will reload only once(timestamp = 0), but I did observe twice case where second reload didn't have timestamp = 0. However, since the update was done in the first reloading, so even the second reloading event enter `addAndGetTotalTokens()`, the `event.newLength - event.oldLength()` is 0, no tokens will be added.

Reply:
>> for all reloading documentChanged events
yes, from what I've observed
- case1: IDE reloading users' change on the document -> timestamp will be 0
- case2: some times IDE will reload the welcome.md (i can't consistent trigger this event), normally it will reload only once(timestamp = 0), but I did observe twice case where second reload didn't have timestamp = 0. However, since the update was done in the first reloading, so even the second reloading event enter `addAndGetTotalTokens()`, the `event.newLength - event.oldLength()` is 0, no tokens will be added.

Reply:
using timestamp will be a more solid approach than what I used previously

Reply:
Loading cwspr welcome.md - 1
![image (1)](https://user-images.githubusercontent.com/96078566/181343132-e4750a6e-e2ba-4bc4-801b-3f7527f34622.png)

Loading cwspr welcome.md - 
![image (2)](https://user-images.githubusercontent.com/96078566/181343353-43c49ca6-7a25-4936-b6b6-079335ea0501.png)
2

Reply:
def add() -> def add(x, y) -> shutdown IDE -> restart IDE
with the current logic, these loading case will pass the condition and get into `addAndGetTotalTokens()`

![image](https://user-images.githubusercontent.com/96078566/181343570-48cadf28-f244-4bc0-9ef7-572d0d89d047.png)

def add() -> (empty) -> shutdown IDE -> restart IDE
![image (1)](https://user-images.githubusercontent.com/96078566/181343840-fc6488eb-0917-4046-9bf4-57c2f437cad1.png)